# Final exercise about functions

Take your time to complete the tasks carefully. :-) Good luck!

### A functional online shop
The math magician wants to convert her online shop to functions. So there's a lot of work waiting for you. 

##### a.) Write a function that calculates the total price of the products in the shopping cart!
Complete the function list_sum(), to which a list with the prices is passed as an argument. The function should then output the sum of the numbers from the list.

In [2]:
cart_prices = [20, 3.5, 6.49, 8.99, 9.99, 14.98]

def list_sum(l):
    total = 0
    for i in l:
        total = total + i
    print(total)
    # or just write: print(sum(l))

list_sum(cart_prices)

63.95


The following output is expected: `63.95`.

##### b.) Write a function that creates a price table for an item!

Now the math magician wants a function to which she can pass an item name and the sales price. From this, the function has to print a list of the sum of one, two, three,... up to ten units prices. Each element in the list should look like this: "Number x articles: Price".

You're just wondering about the math magician's demands for short time.

In [3]:
def prices_list(name, price):
    l = []
    for i in range(1, 11):
        l.append(str(i) + " x " + name + ": " + str(price * i))
    return l

print(prices_list("Cookie", 0.79))

['1 x Cookie: 0.79', '2 x Cookie: 1.58', '3 x Cookie: 2.37', '4 x Cookie: 3.16', '5 x Cookie: 3.95', '6 x Cookie: 4.74', '7 x Cookie: 5.53', '8 x Cookie: 6.32', '9 x Cookie: 7.11', '10 x Cookie: 7.9']


The following output is expected (does not have to have colors):

```python
1 x Cookies: 0.79', '2 x Cookies: 1.58', '3 x Cookies': 2.37.', '4 x Cookies: 3.16', '5 x Cookies: 3.95.', '6 x Cookies: 4.74', '7 x Cookies: 5.53', '8 x Cookies: 6.32', '9 x Cookies: 7.11', '10 x Cookies: 7.9']
```

##### c.) Write a function that fills the lists with the articles!

From now on, a function should also put the goods on the virtual shelves, i.e. put them in the first, still empty position in the _shelf_ list. The article should be passed to the function `add_shelf()` as argument. The function then updates the `shelf` list, and the new article was placed in the first empty shelf. 

In [5]:
shelf = ["Magic Mirror", "empty", "Cookie", "Book with Magic Tricks", "empty"]

def add_shelf(article):
    for i in range(0, len(shelf)):
        if shelf[i] == "empty":
            shelf[i] = article
            break
    
add_shelf("Rubik's Cube")
print(shelf)

['Magic Mirror', "Rubik's Cube", 'Cookie', 'Book with Magic Tricks', 'empty']


The following output is expected (need not be colored):

```python
['Magic Mirror',"Rubik's Cube",'Cookie','Book with Magic Tricks','empty']
```

### Good work :-)